protocol/messenger_handler.go

```diff
@@ -418,6 +418,19 @@ func (m *Messenger) HandleSyncInstallationContact(state *ReceivedMessageState, m
 	return nil
 }
 
+func (m *Messenger) HandleSyncProfilePicture(state *ReceivedMessageState, message protobuf.SyncProfilePicture) error {
```

Comment:
This function should have a clock that comes from `(m *Messenger) getTimesource()`, see usage of `(m *Messenger) getLastClockWithRelatedChat()`.

Once you have the clock value you can compare the time the images were generated against the time of the images in the database, this will mean that only the most recent image changes will be saved on a receiving device.

Reply:
Added `Clock` field to `identity_images` table.